### DISCUSSION TITLE
Auto-restart - how?

### DISCUSSION THREAD

**Discussion by OpenSpacesAndPlaces** [dotnet/yarp#1467] [Q&A] [answered]
If I used IIS, for example, there is an auto-restart on app-crash.

How can I configure the YARP settings to operate the same way?

**Comment by Tratcher** [comment, marked as answer]
YARP doesn't control the process lifecycle, it's a library within your process.

You can use YARP in IIS and you'll get the same auto-restart feature. If you use a console app with Kestrel or Http.Sys as your server then you'll need something to monitor the process and restart it, such as running it as a Windows Service.
